client/state/data-layer/wpcom-http/pipeline/retry-on-failure/test/index.js

```diff
@@ -24,128 +26,121 @@ const withRetries = ( retryCount ) => ( actionOrInbound ) =>
 
 		jest.advanceTimersByTime( 20000 );
-		expect( dispatch ).not.toBeCalled();
+		expect( inbound.store.dispatch ).not.toBeCalled();
 	} );
 
 	test( 'should pass through POST requests', () => {
 		const originalRequest = { ...getSites, method: 'POST' };
-		const inbound = { nextError, originalRequest, store };
+		const inbound = { nextError, originalRequest, store: createMockStore() };
 
 		expect( retryOnFailure( inbound ) ).toEqual( inbound );
 
 		jest.advanceTimersByTime( 20000 );
-		expect( dispatch ).not.toBeCalled();
+		expect( inbound.store.dispatch ).not.toBeCalled();
 	} );
 
 	test( 'should requeue a plain failed request', () => {
-		const inbound = { nextError, originalRequest: getSites, store };
+		const inbound = { nextError, originalRequest: getSites, store: createMockStore() };
 
 		expect( retryWithDelay( 1337 )( inbound ) ).toHaveProperty( 'shouldAbort', true );
-		expect( dispatch ).not.toBeCalled();
+		expect( inbound.store.dispatch ).not.toBeCalled();
 
 		jest.advanceTimersByTime( 1337 );
-		expect( dispatch ).toBeCalledWith( withRetries( 1 )( getSites ) );
+		expect( inbound.store.dispatch ).toBeCalledWith( withRetries( 1 )( getSites ) );
 	} );
 
 	test( 'should requeue only up to `maxAttempts`', () => {
 		const originalRequest = { ...getSites, options: { retryPolicy: { maxAttempts: 3 } } };
-		const inbound = { nextError, originalRequest, store };
+		const inbound = { nextError, originalRequest, store: createMockStore() };
 		const retryIt = retryWithDelay( 1337 );
 
 		// retry 1
 		expect( retryIt( inbound ) ).toHaveProperty( 'shouldAbort', true );
-		expect( dispatch ).not.toBeCalled();
+		expect( inbound.store.dispatch ).not.toBeCalled();
 
 		jest.advanceTimersByTime( 1337 );
-		expect( dispatch ).toBeCalledWith( withRetries( 1 )( originalRequest ) );
+		expect( inbound.store.dispatch ).toBeCalledWith( withRetries( 1 )( originalRequest ) );
 
 		// retry 2
 		expect(
-			retryIt( { ...inbound, originalRequest: dispatch.mock.lastCall[ 0 ] } )
+			retryIt( { ...inbound, originalRequest: inbound.store.dispatch.mock.lastCall[ 0 ] } )
 		).toHaveProperty( 'shouldAbort', true );
-		expect( dispatch.mock.calls.length ).toEqual( 1 );
+		expect( inbound.store.dispatch.mock.calls.length ).toEqual( 1 );
 
 		jest.advanceTimersByTime( 1337 );
-		expect( dispatch.mock.calls.length ).toEqual( 2 );
-		expect( dispatch ).toBeCalledWith( withRetries( 2 )( originalRequest ) );
+		expect( inbound.store.dispatch.mock.calls.length ).toEqual( 2 );
+		expect( inbound.store.dispatch ).toBeCalledWith( withRetries( 2 )( originalRequest ) );
 
 		// retry 3
 		expect(
-			retryIt( { ...inbound, originalRequest: dispatch.mock.lastCall[ 0 ] } )
+			retryIt( { ...inbound, originalRequest: inbound.store.dispatch.mock.lastCall[ 0 ] } )
 		).toHaveProperty( 'shouldAbort', true );
-		expect( dispatch.mock.calls.length ).toEqual( 2 );
+		expect( inbound.store.dispatch.mock.calls.length ).toEqual( 2 );
```

Comment:
Unless I miss some subtle detail, `mock.calls.length` is exactly the same as `.toBeCalledTimes( x )`, would be nice to use the dedicated method.

Looking directly at `mock.calls` should be needed only when checking for some special condition.

Reply:
Yep! You're not missing anything. It's just the consequence of speedy migration 😅

Reply:
Addressed in https://github.com/Automattic/wp-calypso/pull/63862/commits/3ea450d170be6f4b1c1bdad30c2a17cb1ae47c18